scenario_player/utils/reclaim.py

```diff
@@ -0,0 +1,425 @@
+        token = CustomToken(client, token_address, contract_manager, "latest")
+
+        balance = token.balance_of(to_canonical_address(node.address))
+        log.debug(
+            "balance",
+            token=to_checksum_address(token_address),
+            balance=balance,
+            address=node.address,
+        )
+        if balance > 0:
+            drain_amount = balance
+            log.info(
+                "Reclaiming tokens",
+                from_address=node.address,
+                amount=drain_amount.__format__(",d"),
+            )
+            reclaim_amount += drain_amount
+            assert account.address
+            try:
+                token.transfer(account.address, TokenAmount(drain_amount))
+            except InsufficientEth:
+                log.warning(
+                    "Not sufficient eth in node wallet to reclaim",
+                    address=node.address,
+                    token_address=token_address,
+                )
+                continue
+
+    if reclaim_amount:
+        log.info(
+            "Reclaimed",
+            reclaim_amount=reclaim_amount.__format__(",d"),
+            token_address=token.address,
+        )
+
+
+def reclaim_eth(
+    reclamation_candidates: List[ReclamationCandidate], account: Account, eth_rpc_endpoint: URI
+):
+    web3 = Web3(HTTPProvider(eth_rpc_endpoint))
+
+    txs = []
+    reclaim_amount = 0
+    gas_price = web3.eth.gasPrice
+    reclaim_tx_cost = gas_price * VALUE_TX_GAS_COST
+
+    log.info("Checking chain for claimable ETH")
+    for node in reclamation_candidates:
+        balance = web3.eth.getBalance(node.address)
+        if balance > reclaim_tx_cost:
+            drain_amount = balance - reclaim_tx_cost
+            log.info("Reclaiming", from_address=node.address, amount=drain_amount.__format__(",d"))
+            reclaim_amount += drain_amount
+            client = JSONRPCClient(web3, node.privkey)
+            assert account.address
+            txs.append(
+                client.transact(
+                    EthTransfer(
+                        to_address=account.address, value=drain_amount, gas_price=gas_price
+                    )
+                )
+            )
+        (node.node_dir / "reclaimed").touch()
+
+    wait_for_txs(web3, txs, 1000)
+    log.info("Reclaimed", reclaim_amount=reclaim_amount.__format__(",d"))
+
+
+def _get_channels(
+    candidate: ReclamationCandidate,
+    contract_manager: ContractManager,
+    web3: Web3,
+    token_network_address: TokenNetworkAddress,
+    deploy: DeployedContracts,
+) -> List[dict]:
+    """ Read ChannelOpened events to get the channel_ids and participants """
+    new_channel_abi = contract_manager.get_event_abi(CONTRACT_TOKEN_NETWORK, ChannelEvent.OPENED)
+    topics = [
+        encode_hex(event_abi_to_log_topic(new_channel_abi)),  # type: ignore
+        None,
+        encode_hex(bytes([0] * 12) + to_canonical_address(candidate.address)),
```

Comment:
The topics are not set correctly, there is not guarantee that `candidate.address` will be the third fields instead of the second:

https://github.com/raiden-network/raiden-contracts/blob/d3becfb354e250979f1fd81b8966613eb8dd9ded/raiden_contracts/data/source/raiden/TokenNetwork.sol#L301-L306

This either requires two calls, one with the topics `[channel_open_abi, None, address]` and another for `[channel_open_abi, address, None]`, or the filter should just be  `[channel_open_abi]` and the events have to be filtered locally.  If you do filter locally, consider removing the filtering altogether, since this will be doing a linear search there is no guarantee this will be faster to process anyways, the only benefit it will have is reducing the response size.